rtb/datacache/geo_ad_entity.hpp

```diff
@@ -0,0 +1,91 @@
+/* 
+ * File:   geo_ad.hpp
+ * Author: arseny.bushev@gmail.com
+ *
+ * Created on 15 февраля 2017 г., 1:05
+ */
+
+#ifndef __IPC_DATA_AD_GEO_ENTITY_HPP__
+#define __IPC_DATA_AD_GEO_ENTITY_HPP__
+
+namespace ipc { namespace data {
+    
+    template <typename Alloc>
+    struct geo_ad_entity : base_entity<Alloc>
+    {
+        using char_string = typename base_entity<Alloc>::char_string;
+        using base_type   = base_entity<Alloc>; 
+      
+        //for tagging in multi_index_container
+        
+        struct geo_ad_tag {}; // search on geo_id -> ad_id
+        struct ad_id_tag {}; // search on ad_id
+        struct geo_id_tag {}; // search on geo_id
+        
+       
+        geo_ad_entity( const Alloc & a ) :
+            base_entity<Alloc>(a),
```

Comment:
```
geo_id{},
ad_id{a}
```
I suggest we always try to initialize primitives , as they are skipped by compiler during construction.